File: Day1/solution.py
import sys

data = []

for line in sys.stdin:
    data.append(line)
'''
'''

floor = 0
ind = -1
for index in range(len(data[0])):
	if(data[0][index] is '('):
		floor += 1
	else:
		floor -= 1
	if(floor == -1 and ind == -1):
		ind = index + 1

print floor,
print ind
assert floor == 280
assert ind == 1797

File: Day1/solution.rb
#!/usr/bin/env ruby -i

ind = -1
count = 0
floor = 0
STDIN.read.split("\n").each do |line|
    line.each_char do |elem|
        if elem == '('
            floor+=1
        else
            floor-=1
        end

        if floor == -1 and ind == -1
            ind = count+1
        end
        count += 1
    end
end
print floor
print " "
print ind
puts ""
raise "This is wrong" unless floor == 280